examples/dwarfdump.rs

```diff
@@ -315,22 +373,23 @@ fn dump_entries<R: Reader>(offset: R::Offset,
         }
 
         let mut attrs = entry.attrs();
-        while let Some(attr) = attrs.next().expect("Should parse attribute OK") {
+        while let Some(attr) = attrs.next()? {
             print!("{:indent$}{:27} ", "", attr.name(), indent = indent + 18);
             if flags.raw {
                 println!("{:?}", attr.raw_value());
             } else {
-                dump_attr_value(&attr, &unit, debug_loc, debug_ranges, debug_str);
+                dump_attr_value(&attr, &unit, debug_loc, debug_ranges, debug_str)?;
```

Comment:
We can continue dumping if the iterator doesn't need the result of the operation to find the next item. That's going to need a bit of knowledge of the internals to figure out though. The main other spot that I think is useful are the loops for `debug_info.units()` and `debug_types.units()`. There's possibly a couple more similar to these (anything that uses `parse_initial_length` internally) but it's not worth spending a lot of time on.